srv/salt/_modules/osd.py

```diff
@@ -481,14 +481,23 @@ def quiescent(self):
         last = []
         if self.settings['delay'] == 0:
             raise ValueError("The delay cannot be 0")
+        message = "Unkown error"
         while i < self.settings['timeout']/self.settings['delay']:
             current = self.pg_states()
             if not current:
                 log.warning("PGs are not present")
-                return False
-            if len(current) == 1 and current[0]['name'] == 'active+clean':
-                log.warning("PGs are active+clean")
-                return True
+                return {"result": False, "message": "PGs are not present"}
+            if len(current) == 1:
+                if current[0]['name'] == 'active+clean':
+                    log.warning("PGs are active+clean")
+                    return {"result": True, "message": "PGs are active+clean"}
+                else:
+                    if 'scrub' in current[0]['name']:
+                        message = ("PGs are scrubbing, "
```

Comment:
I think this check needs to be moved out of the `if len(current) == 1` block. Instead, if we want to notify if PGs are scrubbing, we probably have to check every element of the `current` array to see if it contains the string "scrub" (some pgs might be active+clean, some might be scrubbing, some might be undersized, some might be in some other state, and for each one of those states, there'll be another element in the `current` array)